### PR TITLE
[sparkle] - feat(sparkle): introduce DataTable

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR aims at introducing a new `sparkle` component `DataTable`. It is in charge of rendering data as a table (similar to what's currently done with markdown).

This is the first building stone for enhanced workspace analytics:
- [Figma](https://www.figma.com/proto/QOxHwppG64GtPocpDhS6Y7/Product?page-id=7084%3A13757&node-id=7727-92009&viewport=933%2C-113%2C0.32&t=6zqjj6FDxeePd7W6-1&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=7727%3A92009&show-proto-sidebar=1)
- [GitHub issue](https://github.com/dust-tt/tasks/issues/897)

## Risk

None

## Deploy Plan

Deploy `sparkle`

### CODE REVIEW

File: sparkle/src/components/DataTable.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+interface DataTableProps {
+  name?: string;
+  columns: string[];
+  rows: (string | number)[][];
+  enableCopy?: boolean;
+  showLimit?: number;
+  downLoadButton?: ReactNode;
+}
+
+export const DataTable = React.forwardRef<HTMLTableElement, DataTableProps>(
```
Comment (flvndvd) [inline sparkle/src/components/DataTable.tsx:16]: 😍

File: sparkle/src/components/DataTable.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+import { ClipboardCheckIcon, ClipboardIcon, IconButton } from "@sparkle/index";
+import { useCopyToClipboard } from "@sparkle/lib/utils";
+
+interface DataTableProps {
+  name?: string;
+  columns: string[];
+  rows: (string | number)[][];
+  enableCopy?: boolean;
+  showLimit?: number;
+  downLoadButton?: ReactNode;
```
Comment (flvndvd) [inline sparkle/src/components/DataTable.tsx:13]:
```suggestion
  downloadButton?: ReactNode;
```

File: sparkle/src/components/DataTable.tsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+      downLoadButton,
+    }: DataTableProps,
+    ref
+  ) => {
+    const [isCopied, copyToClipboard] = useCopyToClipboard
+      ? useCopyToClipboard()
+      : [false, async () => false];
+
+    const handleCopyTable = async () => {
+      function convertToCSV(
```
Comment (flvndvd) [inline sparkle/src/components/DataTable.tsx:33]: Can you extract from here?

File: sparkle/src/components/DataTable.tsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
+import React, { ReactNode } from "react";
+
+import { Page } from "@sparkle/index";
+import { ClipboardCheckIcon, ClipboardIcon, IconButton } from "@sparkle/index";
+import { useCopyToClipboard } from "@sparkle/lib/utils";
+
+interface DataTableProps {
+  name?: string;
+  columns: string[];
+  rows: (string | number)[][];
```
Comment (flvndvd) [inline sparkle/src/components/DataTable.tsx:10]: Let's create a union type for `SupportedRowsType`

File: sparkle/src/components/DataTable.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+      name,
+      columns,
+      rows,
+      enableCopy,
+      showLimit,
+      downLoadButton,
+    }: DataTableProps,
+    ref
+  ) => {
+    const [isCopied, copyToClipboard] = useCopyToClipboard
```
Comment (flvndvd) [inline sparkle/src/components/DataTable.tsx:28]: Remember hooks consistency! You can't have conditional around hooks. In this case it's fine to always iniatialize.

File: sparkle/src/components/DataTable.tsx [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+      }
+    };
+
+    const maxRows = showLimit ?? -1;
+
+    return (
+      <div className="s-mt-2 s-flex s-flex-col s-gap-2">
+        <div className="s-flex s-w-full s-items-center s-justify-between s-gap-1">
+          {name && <Page.H variant="h6">{name}</Page.H>}
+          {downLoadButton && downLoadButton}
```
Comment (flvndvd) [inline sparkle/src/components/DataTable.tsx:59]:
```suggestion
          {downLoadButton}
```

**Review by Duncid — APPROVED** [review]

**Review by flvndvd — APPROVED** [review]

**Pull request merged**